### PR TITLE
[process-agent] [system-probe] Accept multiple API keys per endpoint

### PR DESCRIPTION
**Pull request by robertjli**
### What does this PR do?

Fixes a bug where endpoints with multiple API keys would only send the last key to the forwarder.

### Motivation

I found a bug 🐛 

### Describe your test plan

Check forwarder.go startup log to verify that endpoints reflect the correct number of keys.

### CODE REVIEW

File: cmd/process-agent/collector.go [hunk 1]
```diff
@@ -398,7 +398,12 @@ func keysPerDomains(endpoints []api.Endpoint) map[string][]string {
 	keysPerDomains := make(map[string][]string)
 
 	for _, ep := range endpoints {
-		keysPerDomains[removePathIfPresent(ep.Endpoint)] = []string{ep.APIKey}
+		domain := removePathIfPresent(ep.Endpoint)
+		if keys, ok := keysPerDomains[domain]; ok {
+			keysPerDomains[domain] = append(keys, ep.APIKey)
+		} else {
+			keysPerDomains[domain] = []string{ep.APIKey}
+		}
```
Comment (shang-wang) [inline cmd/process-agent/collector.go:406]: nit: I don't think you need the `if else` statement and just append?

Comment (robertjli) [reply cmd/process-agent/collector.go:406]: TIL 🙇

**Review by shang-wang — DISMISSED** [review]

File: cmd/process-agent/collector.go (hunk 1 above)
Comment (robertjli) [reply cmd/process-agent/collector.go]: Done!

**Review by shang-wang — APPROVED** [review]

**Comment by kevinconaway** [comment]
Will you please write an API test for this behavior as well?

**Pull request merged**